src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReplacementService.cs

```diff
@@ -141,29 +135,50 @@ await plan.Footnotes
 
                     await ReplaceReleaseSubject(releaseId, originalSubjectId, replacementSubjectId);
 
+                    // Replace data guidance
+                    replacementReleaseFile.Summary = originalReleaseFile.Summary;
+
                     await _contentDbContext.SaveChangesAsync();
                     await _statisticsDbContext.SaveChangesAsync();
 
                     return await RemoveOriginalSubjectAndFileFromRelease(releaseId, originalFileId, replacementFileId);
                 });
         }
 
-        private async Task<Either<ActionResult, File>> CheckFileExists(Guid releaseId, Guid fileId)
+        private async Task<Either<ActionResult, (ReleaseFile originalReleaseFile, ReleaseFile replacementReleaseFile)>>
+            CheckReleaseFilesExist(
+                Guid releaseId,
+                Guid originalFileId,
+                Guid replacementFileId)
         {
-            return await _contentPersistenceHelper.CheckEntityExists<ReleaseFile>(
-                    q => q.Include(rf => rf.File)
-                        .Where(rf => rf.ReleaseId == releaseId && rf.FileId == fileId)
-                )
-                .OnSuccess<ActionResult, ReleaseFile, File>(releaseFile =>
+            return await _contentDbContext.ReleaseFiles
+                .Include(rf => rf.File)
+                .FirstOrNotFoundAsync(rf => rf.ReleaseId == releaseId
+                                            && rf.FileId == originalFileId
+                                            && rf.File.Type == FileType.Data)
+                .OnSuccessCombineWith(async _ => await _contentDbContext.ReleaseFiles
+                    .Include(rf => rf.File)
+                    .FirstOrNotFoundAsync(rf => rf.ReleaseId == releaseId
+                                                && rf.FileId == replacementFileId
+                                                && rf.File.Type == FileType.Data))
```

Comment:
I was looking at this and wondering whether it could be a single query - not sure it's worth the bother, cause I think it require extra code below and performance isn't a concern here. Honestly, wouldn't be surprised if you tried one query first and then switched to this :)

Reply:
I already considered it but decided against it because I figured we would have to handle the case where the resultset of the database query contained zero items or only 1 item.

Even in the case where both items are returned as expected we would still have needed to pick out the original and replacement items by file id to return them from this method in a consistent order since that wouldn't be guaranteed by the database query.